doc/1.4/language.md

```diff
@@ -3395,116 +3395,129 @@ the implementations are provided by hierarchically unrelated templates such that
+```
+template gated_write is write {
+    method write_allowed(uint64 val) -> (bool) default {
         return true;
     }
 
-    method on_write_attempted_when_not_allowed() default {
-        log spec_viol: "%s was written to when not allowed", qname;
+    method write(uint64 val) default {
+        if (write_allowed(val)) {
+            base_write_of_gated_write(val); // instead of default()
+        }
     }
 
-    method write_register(uint64 written, uint64 enabled_bytes,
-                          void *aux) default {
-        if (write_allowed()) {
-            default(written, enabled_bytes, aux);
-        } else {
-            on_write_attempted_when_not_allowed();
-        }
+    method base_write_of_gated_write(uint64 val) default {
+        // This is the implementation that default() would've resolved to
+        this.templates.write.write(val);
     }
 }
 
-template very_complex_gated_register is (very_complex_register,
-                                         gated_register) {
-    // No sensible way to combine the two implementations by calling both.
-    // Even if there were, calling both implementations would cause each field
-    // of the register to be written to multiple times, potentially duplicating
-    // side-effects, which is undesirable.
-    // Instead, very_complex_register is chosen as the base implementation
-    // called, and the behaviour of gated_register is replicated around that
-    // call.
-    method write_register(uint64 written, uint64 enabled_bytes,
-                          void *aux) default {
-        if (write_allowed()) {
-            this.templates.very_complex_register.write_register(
-                written, enabled_bytes, aux);
-        } else {
-            on_write_attempted_when_not_allowed();
-        }
-    }
+template write_1_clears is (write, get) { // unchanged
+    ...
 }
 
-in each (gated_register, very_complex_register) {
-    is very_complex_gated_register;
+template gated_write_1_clears is (gated_write, write_1_clears) {
+    // Resolve the conflict by providing an unambiguously most specific
+    // write implementation
+    method write(uint64 val) default {
+        // This makes gated_write the first link in the call chain...
+        this.templates.gated_write.write(val);
+    }
+
+    // ... and by overriding gated_write's base method, we make write_1_clears
+    // the second (and final) link in the chain
+    method base_write_of_gated_write(uint64 val) default {
+        this.templates.write_1_clears.write(val);
+    }
 }
+
+in each (gated_write, write_1_clears) { is gated_write_1_clears; }
 ```
+> [!NOTE]
+> The example given above applies regardless of whether the method
+> implementations in the original templates are `shared` or not. However, the
+> implementations in the template defined to resolve the conflicts might need to
+> be non-`shared` if some of the implementations involved are not `shared`;
+> see the final paragraph of this subsection.
```

Comment:
The added value of this note is questionable -- I expect users would read it as "as usual, you sometimes need to add a `shared` annotation; the compiler will let you know". This is already true for all method declarations.